simpleviewinc/keg-cli
# Updates

**Ticket**: [ZEN-301](https://jira.simpleviewtools.com/browse/ZEN-301)

## Context

* To make working with pull request easier, the keg-cli should allow create and pushing copied images of a developers currently working ENV to github's package repository
* The devs testing the pull request, would then run just one keg-cli task locally
* This command would pull down the pushed package
* It would then run the pushed package locally which would be a complete copy of the original developers ENV


## Goal

* The goal is to create a docker image the matches the **exact environment** in which the container was created
* This allows team members to share code and applications quickly and easily without out the worry of environment difference between developer machines or setups
  * The only real requirement is that the machine can run a docker container
  * Installing the [Keg-CLI](https://github.com/simpleviewinc/keg-cli) is **NOT** required but helpful
    * It automates most of the steps, but technically everything can be done without it

## Updates
* Added packages.md file to explain how packages work 

## Testing

* Pull down the latest changes on master
  * Run the command `keg config sync` to sync you're local global config with any updates
    * To validate it's synced correctly run the command `keg config print`
    * This will print your global config to the console
    * Find the docker.namespace field
      * It should have the value of `simpleviewinc/keg-packages`
      * Looks like [this](http://snpy.in/8o9dnM)
* Next, make sure you can log into the docker-provider
  * Run command => `keg docker provider login` || `keg dp login`
    * This command should auto-login assuming your global config is setup correctly
* Finally, once you are logged in, Read over and follow the steps in [docs/packages.md](https://github.com/simpleviewinc/keg-cli/blob/develop/docs/packages.md)
  * Its the same file that's included in this pull request
